github.js

```diff
@@ -199,20 +148,14 @@ function configureCredentials(config, ui) {
 
   return Promise.resolve()
   .then(function() {
-    ui.log('info', 'If using two-factor authentication or to avoid using your password you can generate an access token at %https://' + (config.hostname || 'github.com') + '/settings/tokens%. Ensure it has `public_repo` scope access.');
-    return ui.input('Enter your GitHub username');
+    ui.log('info', 'You can generate an access token at %https://' + (config.hostname || 'github.com') + '/settings/tokens%.');
+    return ui.input('Enter your GitHub access token');
   })
-  .then(function(username) {
-    auth.username = username;
-    if (auth.username)
-      return ui.input('Enter your GitHub password or access token', null, true);
-  })
-  .then(function(password) {
-    auth.password = password;
-    if (!auth.username)
-      return false;
-
-    return ui.confirm('Would you like to test these credentials?', true);
```

Comment:
Why? That means none of this can ever be deprecated or removed. We want to make sure people's existing configuration works, but why does that mean we need to support both forms forever?

That also creates a more confusing situation with the access tokens - if you use username/password you need to give public_repo, and otherwise you don't. Doesn't seem great